src/eckit/io/PooledFile.cc

```diff
@@ -262,3 +275,21 @@ PooledFileError::PooledFileError(const std::string& file, const std::string& msg
     FileError(msg + " : error on pooled file " + file, loc) {}
 
 }  // namespace eckit
+
+namespace {
+
+eckit::PoolFileEntry* Pool::get(const eckit::PathName& name) {
+    std::lock_guard<std::mutex> lock(filePoolMutex_);
+    auto j = filePool_.find(name);
+    if (j == filePool_.end()) {
+        filePool_.emplace(std::make_pair(name, std::unique_ptr<eckit::PoolFileEntry>(new eckit::PoolFileEntry(name))));
```

Comment:
remove the std::make_pair, you can emplace(key, value) directly avoiding a (actualy, more than one) temporary